src/main/java/org/folio/innreach/domain/processor/ItemShippedInnReachCirculationProcessor.java

```diff
@@ -0,0 +1,67 @@
+package org.folio.innreach.domain.processor;
+
+import java.util.Objects;
+import java.util.UUID;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import org.folio.innreach.client.InventoryClient;
+import org.folio.innreach.domain.CirculationOperation;
+import org.folio.innreach.domain.entity.InnReachTransaction;
+import org.folio.innreach.domain.entity.TransactionPatronHold;
+import org.folio.innreach.domain.exception.EntityNotFoundException;
+import org.folio.innreach.dto.InnReachResponseDTO;
+import org.folio.innreach.dto.TransactionHoldDTO;
+import org.folio.innreach.repository.InnReachTransactionRepository;
+
+@Log4j2
+@Service
+@RequiredArgsConstructor
+public class ItemShippedInnReachCirculationProcessor implements InnReachCirculationProcessor {
+
+  private final InnReachTransactionRepository transactionRepository;
+  private final InventoryClient inventoryClient;
+
+  @Override
+  public boolean canProcess(String circulationOperationName) {
+    return CirculationOperation.ITEM_SHIPPED.getOperationName().equals(circulationOperationName);
+  }
+
+  @Override
+  @Transactional
+  public InnReachResponseDTO process(String trackingId, String centralCode, TransactionHoldDTO transactionHold) {
```

Comment:
According to spec "itemshipped" request can contain only 2 fields:
- callNumber
- itemBarcode

TransactionHoldDTO, according to its schema, has several required fields:
`"required": [
    "transactionTime",
    "pickupLocation",
    "patronId",
    "patronAgencyCode",
    "itemAgencyCode",
    "itemId"
  ]`
Non of these fields are present in the request. How is it expected to make conversion from the request into the object if the mandatory fields are absent?

Reply:
@dmtkachenko yes, my fault. I suppose we could use a single DTO to handle all types of request requests, in this case we need to remove all fields from the "required" block to make them "optional".

Reply:
or we could use generic dto `CirculationRequestDTO` without required fields and then map it to more specific DTO  (i.e. `TransactionHoldDTO`) where needed applying validation in runtime with `javax.validation.Validator`